airflow/models.py

```diff
@@ -920,6 +920,17 @@ def is_queueable(
         else:
             return False
 
+
+    def is_premature(self):
```

Comment:
At first I thought you could use `not ready_for_retry()` wherever `is_premature()` is used, but that's actually different, which validates the existence of  `is_premature()`.